File: src/main/java/ar/edu/um/haberes/rest/model/view/CargoClaseDetallePeriodo.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model.view;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import org.hibernate.annotations.Immutable;

import ar.edu.um.haberes.rest.model.Auditable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "vw_cargo_clase_detalle_periodo")
@Immutable
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class CargoClaseDetallePeriodo extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -2668207560384966175L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cargoClaseDetalleId;
	
	private Long legajoId;
	private Integer anho;
	private Integer mes;
	private Long cargoClaseId;
	private Integer dependenciaId;
	private Integer facultadId;
	private Integer geograficaId;
	private Integer horas;
	private BigDecimal valorHora;
	private Long cargoClasePeriodoId;
	private Byte liquidado;
	private Long periodo;
	
}

File: src/main/java/um/haberes/rest/kotlin/model/view/CargoClaseDetallePeriodo.kt
package um.haberes.rest.kotlin.model.view

import jakarta.persistence.*
import org.hibernate.annotations.Immutable
import um.haberes.rest.kotlin.model.Auditable
import java.math.BigDecimal

@Entity
@Table(name = "vw_cargo_clase_detalle_periodo")
@Immutable
data class CargoClaseDetallePeriodo(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cargoClaseDetalleId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int? = null,
    var mes: Int? = null,
    var cargoClaseId: Long? = null,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var horas: Int? = null,
    var valorHora: BigDecimal? = null,
    var cargoClasePeriodoId: Long? = null,
    var liquidado: Byte? = null,
    var periodo: Long? = null

) : Auditable()
